pkg/manager/member/tikv_scaler.go

```diff
@@ -231,29 +263,37 @@ func (s *tikvScaler) SyncAutoScalerAnn(meta metav1.Object, actual *apps.Stateful
 	return nil
 }
 
-func (s *tikvScaler) preCheckUpStores(tc *v1alpha1.TidbCluster, podName string) (bool, error) {
+func (s *tikvScaler) preCheckUpStores(tc *v1alpha1.TidbCluster, podName string, storesInfo *pdapi.StoresInfo, deleteStoreCount int) (bool, *pdapi.StoresInfo, error) {
 	if !tc.TiKVBootStrapped() {
 		klog.Infof("TiKV of Cluster %s/%s is not bootstrapped yet, skip pre check when scale in TiKV", tc.Namespace, tc.Name)
-		return true, nil
+		return true, nil, nil
 	}
 
 	pdClient := controller.GetPDClient(s.deps.PDControl, tc)
 	// get the number of stores whose state is up
 	upNumber := 0
 
-	storesInfo, err := pdClient.GetStores()
-	if err != nil {
-		return false, fmt.Errorf("failed to get stores info in TidbCluster %s/%s", tc.GetNamespace(), tc.GetName())
+	info := storesInfo
+	if info == nil {
+		var err error
+		info, err = pdClient.GetStores()
```

Comment:
do we need to re-`GetStores` after `DeleteStore`? if not, can we only `GetStores` once before calling any `scaleInOne`?

Reply:
Yeah we can have a discussion here.

It would be quite annoying to mock the test here if we re-`GetStores` every time. 

And I think it's ok to use the same information we get from pd in a round, and count the store we delete to get `upNumber` so that we can find out if we should scale in when `scaleInOne` with given ordinary.

WDYT?

Reply:
OK. so can we just `GetStore` once before calling any `scaleInOne`?

Reply:
Sure.